resolwe/flow/models/data.py

```diff
@@ -319,31 +322,58 @@ def create_entity(self):
         any `Entity`, create new `Entity`
 
         """
-        ds_slug = self.process.flow_collection  # pylint: disable=no-member
-        if ds_slug:
-            entity_query = Entity.objects.filter(data__in=self.parents.all()).distinct()  # pylint: disable=no-member
-
-            if entity_query.count() == 1:
-                entity = entity_query.first()
-
-                copy_permissions(entity, self)
+        entity_type = self.process.entity_type  # pylint: disable=no-member
+        entity_descriptor_schema = self.process.entity_descriptor_schema  # pylint: disable=no-member
+        entity_input = self.process.entity_input  # pylint: disable=no-member
+
+        if entity_type:
+
+            if entity_input:
+                input_id = dict_dot(self.input, entity_input, default=lambda: None)
+                if input_id is None:
+                    logger.warning("Skipping creation of entity due to missing input.")
+                    return
+                if isinstance(input_id, int):
+                    entity_query = Entity.objects.filter(data__pk=input_id)
+                elif isinstance(input_id, list):
+                    entity_query = Entity.objects.filter(data__pk__in=input_id)
```

Comment:
Filter by `type` too. You may want to only define the data filter within `if entity_input:` and then run the query outside.